Unexpected stat(v)fs struct size on i686

The following two tests fail when running on a 32 bit x86 system.
```
==> Starting check()...
/usr/bin/ruby -I/usr/lib/ruby/gems/3.2.0/gems/rspec-support-3.13.1/lib:/usr/lib/ruby/gems/3.2.0/gems/rspec-core-3.13.0/lib /usr/lib/ruby/gems/3.2.0/gems/rspec-core-3.13.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb
Run options: exclude {:dragonfly=>true, :windows=>true}
............................................................................................FF...

Failures:

  1) Sys::Filesystem FFI statfs struct is expected size
     Failure/Error: expect(Sys::Filesystem::Structs::Statfs.size).to eq(dummy.check_sizeof('struct statfs', header))

       expected: 64
            got: 84

       (compared using ==)
     # ./spec/sys_filesystem_unix_spec.rb:505:in `block (3 levels) in <top (required)>'

  2) Sys::Filesystem FFI statvfs struct is expected size
     Failure/Error: expect(Sys::Filesystem::Structs::Statvfs.size).to eq(dummy.check_sizeof('struct statvfs', 'sys/statvfs.h'))

       expected: 72
            got: 92

       (compared using ==)
     # ./spec/sys_filesystem_unix_spec.rb:509:in `block (3 levels) in <top (required)>'

Finished in 0.38878 seconds (files took 0.11209 seconds to load)
97 examples, 2 failures
```

Arch uses glibc which only provides `struct statfs64` and co with `-D_LARGEFILE64_SOURCE=1`. That said, `feature_test_macros(7)` [on my system states](https://man.archlinux.org/man/feature_test_macros.7.en):

> New  programs should not employ this macro; instead _FILE_OFFSET_BITS=64 should be employed.

Without that define you get
```
test.c:6:47: error: invalid application of ‘sizeof’ to incomplete type ‘struct statfs64’
    6 |     printf("struct statfs size %zu\n", sizeof(struct statfs64));
      |                                               ^~~~~~
```

I also found out that musl provides 64 bit functions only. I don't know if anyone has tried this gem on a musl system but I suspect there it'll expect the 32 bit variants and fail (since it won't find the `64` in defs).

--- a/lib/sys/unix/sys/filesystem/structs.rb
+++ b/lib/sys/unix/sys/filesystem/structs.rb
@@ -227,7 +227,9 @@ module Sys
             :f_spare, [:int, 5]
           )
         else
-          layout(
+          require 'mkmf-lite'
+          extend Mkmf::Lite
+          layout_array = [
             :f_bsize, :ulong,
             :f_frsize, :ulong,
             :f_blocks, :uint64,
@@ -241,7 +243,16 @@ module Sys
             :f_namemax, :ulong,
             :f_type, :uint,
             :f_spare, [:int, 5]
-          )
+          ]
+
+          # Check if we need to add __f_unused
+          f_unused_bytes = 2 * check_sizeof('int') - check_sizeof('long')
+          if f_unused_bytes > 0
+            # For Big Endian, this should be inserted at -9
+            layout_array.insert(-7, :__f_unused, [:char, f_unused_bytes])
+          end
+
+          layout(*layout_array)
         end
       end
 

